src/ORM/DataQuery.php

```diff
@@ -842,6 +842,14 @@ public function limit(?int $limit, int $offset = 0): static
         return $this;
     }
 
+    /**
+     * Get the current limit of this query.
+     */
+    public function getLimit(): array
```

Comment:
This method seems really out of place in the class - there are no other "get*" methods doing the equivalent  getting of something on $this->query.

It also seems like a strange way to enfoce NOT doing something somwhere else

Is there another way to achieve the same thing somewhere e.g. in EagerLoadedList?

Reply:
> It also seems like a strange way to enfoce NOT doing something somwhere else

It's obviously not _directly_ enforcing that, but we do need to get the current `limit` value in order to check if it has been set or not.

> Is there another way to achieve the same thing somewhere e.g. in EagerLoadedList?

I assume you mean in `DataList` - by the time you hit `EagerLoadedList` all of the data has been loaded already, so it's way too late.
Also, not, without reflection there isn't another way to do it that I'm aware of.

Reply:
Remove, this new API since we can avoid adding it, just use `$limit = $fetchList->dataQuery->query()->getLimit();` instead

Reply:
Feels untidy but https://github.com/silverstripe/silverstripe-framework/pull/11140#discussion_r1495066358